src/centralserver/admin-service/src/main/java/org/niis/xroad/centralserver/restapi/openapi/SystemApiController.java

```diff
@@ -49,13 +57,50 @@
 
     private final InitializationService initializationService;
     private final InitializationStatusConverter initializationStatusConverter;
+    private final SystemParameterService systemParameterService;
+    private final AuditDataHelper auditDataHelper;
+    private final HAConfigStatus currentHaConfigStatus;
 
-    @Autowired
-    HAConfigStatus currentHaConfigStatus;
 
     @Override
     @PreAuthorize("hasAuthority('SYSTEM_STATUS')")
     public ResponseEntity<SystemStatus> systemStatus() {
+        return getSystemStatusResponseEntity();
+    }
+
+
+    /**
+     * PUT /system/status/server-address : update the server address
+     *
+     * @param serverAddressUpdateBody New central server address (required)
+     * @return System status with updated Central Server address (status code 200)
+     * or request was invalid (status code 400)
+     * or authentication credentials are missing (status code 401)
+     * or request has been refused (status code 403)
+     * or resource requested does not exist (status code 404)
+     * or request specified an invalid format (status code 406)
+     * or internal server error (status code 500)
+     */
+    @Override
+    @PreAuthorize("hasAuthority('SYSTEM_STATUS')")
+    @AuditEventMethod(event = RestApiAuditEvent.UPDATE_CENTRAL_SERVER_ADDRESS)
+    public ResponseEntity<SystemStatus> updateCentralServerAddress(
+            @ApiParam(value = "New central server address", required = true) @Validated @RequestBody
```

Comment:
Redundant annotations. They are already introduced in the `SystemApi` interface

Reply:
Done in: f29ed4e